src/DnsRootStack.ts

```diff
@@ -46,6 +61,59 @@ export class DnsRootStack extends cdk.Stack {
 
   }
 
+  setupDnsManagementRole() {
+    const role = new IAM.Role(this, 'role', {
+      roleName: 'dns-manager',
+      description: 'Role for dns-management account with access rights to IAM (readonly) and Route53',
+      assumedBy: new IAM.PrincipalWithConditions(
+        new IAM.AccountPrincipal(Statics.iamAccountId), //IAM account
+        {
+          Bool: {
+            'aws:MultiFactorAuthPresent': true,
+          },
+        },
+      ),
+    });
+
+    /**
+     * Policy for route53
+     */
+    role.addToPolicy(
+      new IAM.PolicyStatement({
+        effect: IAM.Effect.ALLOW,
+        actions: [
+          'route53:*', // Allow domain management
+          'route53domains:*', // Allow domain management
+        ],
+        resources: [
+          '*',
+        ],
+      }),
+    );
+
+    /**
+     * Policy for iam
+     */
+    role.addToPolicy(
+      new IAM.PolicyStatement({
+        effect: IAM.Effect.ALLOW,
+        actions: [
+          'iam:Get*',
+          'iam:List*',
+        ],
+        resources: [
+          '*',
```

Comment:
Hier ook?

Reply:
Ook hier zijn nog geen arns bekend, wel namen van de rollen waarvoor het interessant is om ze in te zien.